superset-frontend/src/explore/controlPanels/CalHeatmap.js

```diff
@@ -35,21 +78,94 @@ export default {
       expanded: true,
       controlSetRows: [
         ['linear_color_scheme'],
-        ['cell_size', 'cell_padding'],
-        ['cell_radius', 'steps'],
-        ['y_axis_format', 'x_axis_time_format'],
+        [
+          {
+            name: 'cell_size',
+            config: {
+              type: 'TextControl',
+              isInt: true,
+              default: 10,
+              validators: [v.integer],
+              renderTrigger: true,
+              label: t('Cell Size'),
+              description: t('The size of the square cell, in pixels'),
+            },
+          },
+          {
+            name: 'cell_padding',
+            config: {
+              type: 'TextControl',
+              isInt: true,
+              validators: [v.integer],
+              renderTrigger: true,
+              default: 2,
+              label: t('Cell Padding'),
+              description: t('The distance between cells, in pixels'),
+            },
+          },
+        ],
+        [
+          {
+            name: 'cell_radius',
+            config: {
+              type: 'TextControl',
+              isInt: true,
+              validators: [v.integer],
+              renderTrigger: true,
+              default: 0,
+              label: t('Cell Radius'),
+              description: t('The pixel radius'),
+            },
+          },
+          {
+            name: 'steps',
+            config: {
+              type: 'TextControl',
+              isInt: true,
+              validators: [v.integer],
+              renderTrigger: true,
+              default: 10,
+              label: t('Color Steps'),
+              description: t('The number color "steps"'),
+            },
+          },
+        ],
+        [
+          'y_axis_format',
+          {
+            name: 'x_axis_time_format',
```

Comment:
@kristw it appears I can change this one to use `x_axis_format` with a couple config overrides. This would get rid of `x_axis_time_format` entirely, which appears to be a deprecated practice. 

If that's a good idea, I can make the change here, and a quick PR in `superset-ui-plugins` to match. 

However, I'm not sure if this leads to any painful migrations for existing charts/dashboards. Thoughts?

Reply:
Umm, as much as I wanna merge them, this will cause `db:migration` for existing charts/dashboards so perhaps let's hold on to it.

Reply:
Ok... I'll try to keep track of that, as it may be worth addressing this and other similar migrations... maybe we can get a handful of them in a batch. Heads up to @villebro or @willbarrett that this may come up in discussion.

Reply:
@runsackas I like the idea of doing a migration batch to clean up duplicate controls. Let's hold off on the changes for now and put them in a google sheet or similar so we can do one or two migrations when the control migration is otherwise done.